generate-serializers should fail generation if an attribute isn't recognized

gavin-apple opened the pull request:
#### ea721e9383ae4863df1af7a8810b0bdf5f37eade
<pre>
generate-serializers should fail generation if an attribute isn&apos;t recognized
<a href="https://bugs.webkit.org/show_bug.cgi?id=299185">https://bugs.webkit.org/show_bug.cgi?id=299185</a>
<a href="https://rdar.apple.com/160944292">rdar://160944292</a>

Reviewed by Alex Christensen.

Previously generate-serializers.py would just continue silently if it didn&apos;t recognize
an attribute. This can lead to errors either later in compilation or in the runtime
behavior. Instead, this change cause us to hard error on generation.

* Source/WebKit/Scripts/generate-serializers.py:
(SerializedType.__init__):
* Source/WebKit/Scripts/webkit/tests/Makefile:
* Source/WebKit/Scripts/webkit/tests/TestInvalidAttributes.serialization.in: Added.
* Source/WebKit/Shared/cf/CFTypes.serialization.in:

Canonical link: <a href="https://commits.webkit.org/300314@main">https://commits.webkit.org/300314@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7bb11a661519d656026ac1119e31f019169f791c

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/121848 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/41549 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/32213 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/128399 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/73941 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/d9ba8de5-2595-439e-bc34-458a89261f4b) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/42261 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/50142 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/92596 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/61541 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/7cda23b7-14ab-4127-8e27-9a2be49f17dd) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/124800 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/33703 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/109118 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/73256 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/4a5ccdac-fc98-4ba5-a256-8c92cf76664f) 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/121205 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/32726 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/71901 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/114010 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/103218 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/27467 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/131170 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/120370 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/48783 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/37108 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/101180 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/49155 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/105335 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/101048 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25666 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/46412 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/24517 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/45487 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/48641 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/54382 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/150531 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/48111 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/38514 "Found 1 new JSC stress test failure: stress/dont-link-virtual-calls-on-compiler-thread.js.no-llint (failure)") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/51466 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/49794 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

sheeparegreat reviewed (commented):
Looks good to me.  NIT: You could add a test for the exception throwing like [here](https://github.com/WebKit/WebKit/pull/50901/files#diff-0a6dadb7bef4e6cce95ceccd7d602659df577369345dcee1f1fe3c36f56d8d05)

gavin-apple commented on Source/WebKit/Shared/cf/CFTypes.serialization.in:
```diff
@@ -69,7 +69,7 @@ additional_forward_declaration: typedef struct CF_BRIDGED_TYPE(id) __SecKeychain
 
 additional_forward_declaration: typedef struct CF_BRIDGED_TYPE(id) __CVBuffer *CVPixelBufferRef
-[WebKitPlatform, CustomHeader,  AdditionalEncoder=StreamConnectionEncoder, ToCFMethod=result->toCF()] CVPixelBufferRef wrapped by WebKit::CoreIPCCVPixelBufferRef {
+[WebKitPlatform, CustomHeader, ToCFMethod=result->toCF()] CVPixelBufferRef wrapped by WebKit::CoreIPCCVPixelBufferRef {
```
This attribute was previously misspelled (extra space before) and therefore the StreamEncoder wasn't ever being generated.

gavin-apple committed:
generate-serializers should fail generation if an attribute isn't recognized
https://bugs.webkit.org/show_bug.cgi?id=299185
rdar://160944292

Reviewed by Alex Christensen.

Previously generate-serializers.py would just continue silently if it didn't recognize
an attribute. This can lead to errors either later in compilation or in the runtime
behavior. Instead, this change cause us to hard error on generation.

* Source/WebKit/Scripts/generate-serializers.py:
(SerializedType.__init__):
* Source/WebKit/Scripts/webkit/tests/Makefile:
* Source/WebKit/Scripts/webkit/tests/TestInvalidAttributes.serialization.in: Added.
* Source/WebKit/Shared/cf/CFTypes.serialization.in:

Canonical link: https://commits.webkit.org/300314@main
